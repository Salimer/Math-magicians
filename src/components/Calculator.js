import React, { useState } from 'react';
import '../styles/style.scss';
import calculate from '../logic/calculate';
import OperatorBtns from './OperatorBtns';
import NumbersBtns from './NumbersBtns';

function Calculator() {
  const [calculatorData, setCalculatorData] = useState({
    total: null,
    next: null,
    operation: null,
  });

  const handleButtonClick = (event) => {
    const clicked = event.target.textContent;
    const value = calculate(calculatorData, clicked);
    setCalculatorData(value);
  };

  const { total, operation, next } = calculatorData;
  return (
    <article className="calculator-container">
      <div className="screen">{ next || operation || total || 0 }</div>
      <OperatorBtns handleButtonClick={handleButtonClick} />
      <NumbersBtns handleButtonClick={handleButtonClick} />
    </article>
  );
}

export default Calculator;
