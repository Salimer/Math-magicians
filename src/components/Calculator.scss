.calculator-container {
  display: grid;
  grid-template-areas:
    "screen screen screen screen"
    "numbers numbers numbers operator"
    "numbers numbers numbers operator"
    "numbers numbers numbers operator"
    "numbers numbers numbers operator"
    "numbers numbers numbers operator";

  .screen {
    grid-area: screen;
    color: #fff;
    font-size: 2rem;
    text-align: right;
    padding: 1rem 0.5rem;
    background-color: #858693;
  }

  .numbers {
    grid-area: numbers;
    display: grid;
    grid-template-columns: repeat(3, 1fr);
    background-color: #e0e0e0;

    div:nth-child(13) {
      grid-column: span 2;
    }

    .button {
      text-align: center;
      border: 1px solid #bababa;
      padding: 1rem;
    }
  }

  .operator {
    grid-area: operator;
    display: grid;
    grid-template-columns: repeat(1, 1fr);
    background-color: #f4913e;

    .button {
      text-align: center;
      border: 1px solid #bababa;
      padding: 1rem;
    }
  }
}
