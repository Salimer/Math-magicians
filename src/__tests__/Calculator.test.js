import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect'; // Import extend-expect
import Calculator from '../components/Calculator';

test('renders Calculator correctly', () => {
  const { container } = render(<Calculator />);
  expect(container).toMatchSnapshot();
});

test('updates calculator data correctly when a button is clicked', () => {
    const { getAllByText } = render(<Calculator />);
    fireEvent.click(getAllByText('2')[0]); // Select the first matching element
    expect(getAllByText('2')[0]).toBeInTheDocument();
  });
  
