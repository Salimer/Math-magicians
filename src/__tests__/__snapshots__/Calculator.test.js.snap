// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Calculator correctly 1`] = `
<div>
  <article
    class="calculator-container"
  >
    <div
      class="screen"
    >
      0
    </div>
    <div
      class="operator"
    >
      <button
        class="button"
        type="button"
      >
        รท
      </button>
      <button
        class="button"
        type="button"
      >
        x
      </button>
      <button
        class="button"
        type="button"
      >
        -
      </button>
      <button
        class="button"
        type="button"
      >
        +
      </button>
      <button
        class="button"
        type="button"
      >
        =
      </button>
    </div>
    <div
      class="numbers"
    >
      <button
        class="button"
        type="button"
      >
        AC
      </button>
      <button
        class="button"
        type="button"
      >
        +/-
      </button>
      <button
        class="button"
        type="button"
      >
        %
      </button>
      <button
        class="button"
        type="button"
      >
        7
      </button>
      <button
        class="button"
        type="button"
      >
        8
      </button>
      <button
        class="button"
        type="button"
      >
        9
      </button>
      <button
        class="button"
        type="button"
      >
        4
      </button>
      <button
        class="button"
        type="button"
      >
        5
      </button>
      <button
        class="button"
        type="button"
      >
        6
      </button>
      <button
        class="button"
        type="button"
      >
        1
      </button>
      <button
        class="button"
        type="button"
      >
        2
      </button>
      <button
        class="button"
        type="button"
      >
        3
      </button>
      <button
        class="button"
        type="button"
      >
        0
      </button>
      <button
        class="button"
        type="button"
      >
        .
      </button>
    </div>
  </article>
</div>
`;
